fs/zbd_zenfs.cc

```diff
@@ -174,7 +174,8 @@ Zone *ZonedBlockDevice::GetIOZone(uint64_t offset) {
 
 ZonedBlockDevice::ZonedBlockDevice(std::string bdevname,
                                    std::shared_ptr<Logger> logger)
-    : filename_("/dev/" + bdevname), logger_(logger) {
+    : filename_("/dev/" + bdevname), logger_(logger),
+      metrics_(std::make_shared<NoZenFSMetrics>()) {
```

Comment:
Could we optionally inject a `ZenFSMetrics` instance here? But default to `NoZenFSMetrics`. I don't think it makes sense to change the metrics instance after zbd construction, so we could drop `ZonedBlockDevice::SetMetrics()`.

Reply:
Yes, the user should provide Metrics at initialization time, not SetMetrics at an unknown time. 
I believe it would be much better to add a constructor that can initialize Metrics.

Reply:
I added an initialization function for ZoneBlockDevice to pass in Metrics, and a version for NewZenFS with a Metrics parameter.

Reply:
I don't think we need a new constructor. We can just provide a default argument for the existing constructor:

```suggestion
ZonedBlockDevice::ZonedBlockDevice(std::string bdevname,
                                   std::shared_ptr<Logger> logger,
                                   std::shared_ptr<ZenFSMetrics> metrics = std::make_shared<NoZenFSMetrics>())
    : filename_("/dev/" + bdevname), logger_(logger), metrics_(metrics) {
...
}
```